File: app/src/main/java/art/coded/wireframe/ui/home/HomeFragment.java
package art.coded.wireframe.ui.home;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import art.coded.wireframe.data.Element;
import art.coded.wireframe.databinding.FragmentHomeBinding;

public class HomeFragment extends Fragment {

    static final String LOG_TAG = HomeFragment.class.getSimpleName();

    private HomeViewModel homeViewModel;
    private FragmentHomeBinding binding;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {

        binding = FragmentHomeBinding.inflate(inflater, container, false);
        View root = binding.getRoot();

        homeViewModel =
                new ViewModelProvider(this).get(HomeViewModel.class);

        return root;
    }

    @Override public void onResume() {
        super.onResume();
        final TextView textView = binding.textHome;
        homeViewModel.loadData(requireContext());
        homeViewModel.getText().observe(getViewLifecycleOwner(), o -> textView.setText(o));
    }

    @Override public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/ktx/java/art/coded/wireframe/view/fragment/HomeFragment.kt
package art.coded.wireframe.view.fragment

import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import androidx.lifecycle.ViewModelProvider
import art.coded.wireframe.viewmodel.HomeViewModel
import android.view.View
import androidx.fragment.app.Fragment
import androidx.preference.PreferenceManager
import art.coded.wireframe.R
import art.coded.wireframe.databinding.FragmentHomeBinding
import art.coded.wireframe.model.ElementRepository
import art.coded.wireframe.viewmodel.ListViewModel
import art.coded.wireframe.viewmodel.ListViewModelFactory

private val LOG_TAG = HomeFragment::class.java.simpleName

class HomeFragment : Fragment() {
    private lateinit var homeViewModel: HomeViewModel
    private var binding: FragmentHomeBinding? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        binding = FragmentHomeBinding.inflate(inflater, container, false)
        val root: View = binding!!.root
        val sp = PreferenceManager.getDefaultSharedPreferences(requireContext())
        val key = requireContext().getString(R.string.key_username)
        val default = requireContext().getString(R.string.default_username)
        homeViewModel = ViewModelProvider(
            this,
            HomeViewModel.HomeViewModelFactory(sp, key, default)
        ).get(HomeViewModel::class.java)
        return root
    }

    override fun onResume() {
        super.onResume()
        val textView = binding!!.textHome
        homeViewModel.getData().observe(viewLifecycleOwner) { o: String? -> textView.text = o }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }
}